ext/web/08_text_encoding.js

```diff
@@ -91,40 +94,43 @@
           allowShared: true,
         });
       }
-      options = webidl.converters.TextDecodeOptions(options, {
-        prefix,
-        context: "Argument 2",
-      });
+      let stream = false;
+      if (options !== undefined) {
+        options = webidl.converters.TextDecodeOptions(options, {
+          prefix,
+          context: "Argument 2",
+        });
+        stream = options.stream;
+      }
 
       try {
-        try {
-          if (ArrayBufferIsView(input)) {
-            input = new Uint8Array(
-              input.buffer,
-              input.byteOffset,
-              input.byteLength,
-            );
-          } else {
-            input = new Uint8Array(input);
-          }
-        } catch {
-          // If the buffer is detached, just create a new empty Uint8Array.
+        // Note from spec: implementations are strongly encouraged to use an implementation strategy that avoids this copy.
+        // When doing so they will have to make sure that changes to input do not affect future calls to decode().
+        if (input.byteLength === 0) {
```

Comment:
`TypedArrayPrototypeGetByteLength(input)`
(Or is this polymorpic (`input : ArrayBufferView`)?)

Reply:
`ArrayBufferView | (Shared)ArrayBuffer`

Reply:
This needs a split for `ArrayBufferView`, `TypedArray`, `ArrayBuffer`, `SharedArrayBuffer`.

..I didn't add it to the WebIDL utilities because `SharedArrayBuffer` wasn't in the snapshot back then, is it now?

Reply:
webidl `BufferSource` covnerter checks for SharedArrayBuffer, yes.

Reply:
I mean that the way you're accessing the byte length is polymorphic and accesses a user provided object via `byteLength`, which is strictly incorrect.

Reply:
It needs to dispatch:
```
TypedArray => TypeArrayPrototypeGetByteLength
DataView => DataViewPrototypeGetByteLength
ArrayBuffer => ArrayBufferPrototypeGetByteLength
SharedArrayBuffer => SharedArrayBufferPrototypeGetByteLength
```

Reply:
It needs to dispatch:
```
TypedArray => TypeArrayPrototypeGetByteLength
DataView => DataViewPrototypeGetByteLength
ArrayBuffer => ArrayBufferPrototypeGetByteLength
SharedArrayBuffer => SharedArrayBufferPrototypeGetByteLength
```

Reply:
hmm, maybe handling zero length should be done in Rust